File: hackerrank/30 days of code/day 9, recursion 3/recursion.cs
// https://www.hackerrank.com/challenges/30-recursion/problem?isFullScreen=true

using static System.Console;

public class Solution
{
    public static void Main()
    {
        int n = int.Parse(ReadLine());
        WriteLine(Factorial(n));
    }

    private static int Factorial(int n)
    {
        return n == 1 ? 1 : n * Factorial(n - 1);
    }
}

File: hackerrank/30 days of code/day 9, recursion 3/recursion.rs
// https://www.hackerrank.com/challenges/30-recursion/problem?isFullScreen=true

use text_io::read;

fn main() {
    let n: i32 = read!();
    println!("{}", factorial(n));
}

fn factorial(n: i32) -> i32 {
    if n == 1 {
        return 1;
    }
    n * factorial(n - 1)
}
